clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -10635,8 +10635,50 @@ static void getNonTripleBasedSYCLPostLinkOpts(const ToolChain &TC,
                                               const JobAction &JA,
                                               const llvm::opt::ArgList &TCArgs,
                                               ArgStringList &PostLinkArgs) {
+  // FIXME: Due to the limitations of device sanitizer implementation, we need
+  // to always split device code per kernel if device sanitizer is enabled.
+  bool IsDeviceAsanEnabled = [&] {
+    if (Arg *A = TCArgs.getLastArg(options::OPT_fsanitize_EQ,
+                                   options::OPT_fno_sanitize_EQ)) {
+      if (A->getOption().matches(options::OPT_fsanitize_EQ) &&
+          A->getValues().size() == 1) {
+        std::string SanitizeVal = A->getValue();
+        if (SanitizeVal == "address")
+          return true;
+      }
+    } else {
+      // -Xsycl-target-frontend
+      if (llvm::any_of(TCArgs.getAllArgValues(options::OPT_Xsycl_frontend),
+                       [](const std::string &Arg) {
+                         return Arg.find("-fsanitize=address") !=
+                                std::string::npos;
+                       }))
+        return true;
+
+      // -Xsycl-target-frontend=
+      if (llvm::any_of(TCArgs.getAllArgValues(options::OPT_Xsycl_frontend_EQ),
```

Comment:
There is no scrutiny on any specific target as specified with `-Xsycl-target-frontend=<target> <opt>`.  This case is probably not used much (if at all) but we should at least call out the discrepancy in a comment.

Reply:
Updated the comments

Reply:
Apologies - it looks like my original message didn't provide enough detail of what I was looking for.  The issue is that the enabling of the sanitizer is only gated by using `-Xsycl-target-frontend=<target> -fsanitize=address`.  This means no matter what `<target>` value is used, `-fsanitize-address` will be passed to all device compilations.

For example, `-fsycl -fsycl-targets=spir64_gen,spir64_x86_64,spir64 -Xsycl-target-frontend=spir64 -fsanitize=address` will apply to `spir64_gen` and `spir64_x86_64` as well, even if that is not the users intention.

Reply:
@mdtoguchi, thanks for your patiently explaining, it's clear to me. But we have found a more reasonable fix for the original problem. So, we don't need this PR anymore. Anyway, thanks for your comments again.